src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -407,6 +420,29 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
         if (!isNaN(args.value)) {
           this.cumulativeMaxRefundCountUpdates.push(args);
         }
+      } else if (args.key === utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.LITE_CHAIN_ID_INDICES)) {
+        // We need to parse the chain ID indices array from the stringified JSON. However,
+        // the on-chain string has quotes around the array, which will parse our JSON as a
+        // string instead of an array. We need to remove these quotes before parsing.
+        // To be sure, we can check for single quotes, double quotes, and spaces.
+        const chainIndices = JSON.parse(args.value.replace(/['"\s]/g, ""));
+        // Check that the array is valid and that every element is a number.
+        if (!isArrayOf<number>(chainIndices, isPositiveInteger)) {
+          this.logger.warn({ at: "ConfigStore", message: `The array ${chainIndices} is invalid.` });
+          // If not a valid array, skip.
+          continue;
+        }
+        // Let's also check that the array doesn't contain any duplicates.
+        if (new Set(chainIndices).size !== chainIndices.length) {
+          this.logger.warn({
+            at: "ConfigStore",
+            message: `The array ${chainIndices} contains duplicates making it invalid.`,
+          });
+          // If not a valid array, skip.
+          continue;
+        }
+        // If all else passes, we can add this update.
+        this.liteChainIndicesUpdates.push({ ...args, value: chainIndices });
```

Comment:
Should this list be a subset of the enabled chain id list?

Reply:
We could refine it to be this - this kind of check would need to happen after the original loop but this isn't out of scope

Reply:
Will see how we can fit this into the UMIP